truncate_client/src/lil_bits/daily_splash/mod.rs

```diff
@@ -0,0 +1,217 @@
+use epaint::{emath::Align2, vec2, Color32, TextureHandle};
+use instant::Duration;
+use interpolation::Ease;
+use truncate_core::game::Game;
+
+mod graph;
+mod msg_mock;
+
+use eframe::egui::{self, Id, Order, Sense};
+
+use crate::{
+    app_outer::Backchannel,
+    regions::active_game::GameCtx,
+    utils::{
+        daily::{get_stats, DailyStats},
+        text::TextHelper,
+        Theme,
+    },
+};
+
+use self::{graph::DailySplashGraph, msg_mock::ShareMessageMock};
+
+/*
+
+TODOs for the daily splash screen:
+- Vertically center contents, or size the modal to the contents better
+- Add ability to close the splash screen and look at the game
+
+ */
+
+pub struct DailySplashUI {
+    stats: DailyStats,
+    graph: DailySplashGraph,
+    msg_mock: ShareMessageMock,
+    streak_length: usize,
+    win_rate: f32,
+}
+
+impl DailySplashUI {
+    pub fn new(ui: &mut egui::Ui, game: &Game, ctx: &mut GameCtx, current_time: Duration) -> Self {
+        let stats = get_stats();
+
+        let streak_length = stats
+            .days
+            .values()
+            .rev()
+            .enumerate()
+            .find_map(|(streak_length, day)| {
+                if day.attempts.last().map(|a| a.won) == Some(true) {
+                    None
+                } else {
+                    Some(streak_length)
+                }
+            })
+            .unwrap_or_else(|| stats.days.len());
+
+        let win_count = stats
+            .days
+            .values()
+            .filter(|day| day.attempts.last().map(|a| a.won) == Some(true))
+            .count();
+
+        DailySplashUI::seed_animations(ui);
+
+        let graph = DailySplashGraph::new(ui, &stats, current_time);
+        let msg_mock = ShareMessageMock::new(game, ctx, &stats);
+
+        Self {
+            msg_mock,
+            graph,
+            streak_length,
+            win_rate: win_count as f32 / stats.days.len() as f32,
+            stats,
+        }
+    }
+}
+
+#[derive(Hash)]
+enum Anim {
+    Background,
+    ModalPos,
+    Items,
+    Viz,
+}
+
+impl Into<Id> for Anim {
+    fn into(self) -> Id {
+        Id::new("splash").with(self)
+    }
+}
+
+impl DailySplashUI {
+    fn seed_animations(ui: &mut egui::Ui) {
+        DailySplashUI::anim(ui, Anim::Background, 0.0, 0.0);
+        DailySplashUI::anim(ui, Anim::ModalPos, 0.0, 0.0);
+        DailySplashUI::anim(ui, Anim::Items, 0.0, 0.0);
+        DailySplashUI::anim(ui, Anim::Viz, 0.0, 0.0);
+    }
+
+    // Animates to a given value once (can't be retriggered), applying an easing function
+    fn anim(ui: &mut egui::Ui, val: Anim, to: f32, duration: f32) -> f32 {
+        let linear = ui.ctx().animate_value_with_time(
+            val.into(),
+            if to > 0.0 { 1.0 } else { 0.0 },
+            duration,
+        );
+        let eased = linear.calc(interpolation::EaseFunction::QuadraticOut);
+        eased * to
+    }
```

Comment:
At some point it might be nice to pull this out into our own animation manager that can do easing and such